prow/plugins/approve/approve.go

```diff
@@ -371,11 +375,18 @@ func humanAddedApproved(ghc githubClient, log *logrus.Entry, org, repo string, n
 	}
 }
 
-func approvalCommandMatcher(botName string, lgtmActsAsApprove bool) func(*comment) bool {
+func approvalCommandMatcher(botName string, lgtmActsAsApprove bool, reviewActsAsApprove bool) func(*comment) bool {
 	return func(c *comment) bool {
 		if c.Author == botName || isDeprecatedBot(c.Author) {
 			return false
 		}
+
+		// consider reviews in either approved OR requested changes states as
+		// approval commands. Reviews in requested changes states will be
+		// interpreted as cancelled approvals.
+		if reviewActsAsApprove && (c.ReviewState == approvedReviewState || c.ReviewState == requestedChangesReviewState) {
```

Comment:
How do you want to handle my leaving a `/approve cancel` while also leaving an approve state? Looks like here the review state wins?

There was some talk from Cole that he wanted the slash command to take precedence.

Reply:
Actually the `/approve cancel` command wins because it's processed later. I added some tests to demonstrate this and the converse (a review requesting changes with simultaneous `/approve` results in approval). Take a look and let me know if it's still unclear.